Python/jit.c

```diff
@@ -408,11 +408,26 @@ _PyJIT_Compile(_PyExecutorObject *executor, const _PyUOpInstruction *trace, size
     // Loop again to emit the code:
     unsigned char *code = memory;
     unsigned char *data = memory + code_size;
+    {
+        // Compile the trampoline, which handles changing the calling convention
+        // (on platforms where it's not necessary, the trampoline is empty):
```

Comment:
I guess I meant that the trampoline *isn't necessary* on the platforms where the calling convention *isn't available*. I'll update the comment to make that clearer.

I'd rather not call out specific platforms since those comments could rot with time. `_targets.py` should be the source of truth for this sort of stuff, I think.

For example, yeah, the `ghc` calling convention only supports `aarch64`, `i686`, and `x86_64` targets. But we don't enable it on `aarch64`, and soon we'll be on Clang 19 where we'll instead use the `preserve_none` calling convention, which only supports `x86_64` right now (but may get more platforms with time). We might even do more clever things in the future.

So I'd rather have this comment explain the "why" instead of the "when" and "how", which is mostly irrelevent to actually jitting the thing at runtime. :)